src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -97,6 +98,9 @@ public SamReader open(final Path path) {
     /** Set this factory's {@link htsjdk.samtools.SAMRecordFactory} to the provided one, then returns itself. */
     abstract public SamReaderFactory samRecordFactory(final SAMRecordFactory samRecordFactory);
 
+    /** Set this factory's {@link htsjdk.samtools.util.zip.InflaterFactory} to the provided one, then returns itself. */
+    abstract public SamReaderFactory inflaterFactory(final InflaterFactory inflaterFactory);
```

Comment:
Also clarify whether the factory would be used for something like a `.sam.gz` or not.